### PR TITLE
fix: canShare() return false if not allowed to use

### PR DESCRIPTION
**Pull request by marcoscaceres**
For *normative* changes, the following tasks have been completed:

 * [x] [Modified Web platform tests](https://github.com/web-platform-tests/wpt/pull/30868)

Implementation commitment:

 * [x] WebKit (https://bugs.webkit.org/show_bug.cgi?id=214448)
 * [ ] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=1251175)
 * [x] Gecko (https://phabricator.services.mozilla.com/D90834)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/web-share/pull/220.html" title="Last updated on Sep 20, 2021, 10:35 PM UTC (9a32b1d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-share/220/94e7883...9a32b1d.html" title="Last updated on Sep 20, 2021, 10:35 PM UTC (9a32b1d)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -307,6 +307,9 @@ <h3>
             </li>
             <li>If |document| is not [=Document/fully active=], return false.
             </li>
+            <li>If |document| is not <a>allowed to use</a> <a>"web-share"</a>,
+            return `false`.
```
Comment (marcoscaceres) [inline index.html:311]:
```suggestion
            return false.
```

**Comment by saschanaz** [comment]
Do we have WPT for this BTW?

**Comment by marcoscaceres** [comment]
I'll add one. No problem.

**Comment by marcoscaceres** [comment]
Added/fixed test:
https://github.com/web-platform-tests/wpt/pull/30868
 
> BTW I'm not sure what the enum version of canShare be look like, if canShare covers more than just data check 🤔

Happy to build on that aspect incrementally as a followup. 

Irrespective, I think it would be confusing for ".canShare(x)" return `true` and then calling `.share()` to fail. That would not be great. This keeps them in sync.

**Pull request merged**